app/src/common/main.c

```diff
@@ -16,17 +16,42 @@
 ********************************************************************************/
 #include "app_main.h"
 #include "view.h"
+#include "swap.h"
 
 #include <os_io_seproxyhal.h>
 
+static void app_exit(void) {
+    BEGIN_TRY_L(exit) {
+        TRY_L(exit) {
+            os_sched_exit(-1);
+        }
+        FINALLY_L(exit) {
+        }
+    }
+    END_TRY_L(exit);
+}
+
 __attribute__((section(".boot"))) int
-main(void) {
+main(int arg0) {
     // exit critical section
     __asm volatile("cpsie i");
 
     view_init();
```

Comment:
You can't call `view_init` when the app is executed from the app exchange for a swap flow. The app has to run without the GUI being displayed on the screen.

That's why in my implementation of the swap I did something like this : 

```
    if (!arg0) {  // called from dashboard as standalone app
        view_init();
        app_init();
        app_main();
        return 0;
    }

// below should be the code that processes the swap command
```
@carlosala @ftheirs you can propose something else but it has to do basically the same.